xbmc/guilib/GUIFontCache.h

```diff
@@ -134,7 +134,7 @@ struct CGUIFontCacheKeysMatch
 template<class Position, class Value>
 class CGUIFontCache
 {
-  CGUIFontCacheImpl<Position, Value>* m_impl;
+  std::unique_ptr<CGUIFontCacheImpl<Position, Value>> m_impl;
```

Comment:
Does this really work? While using `std::unique_ptr<>` with incomplete types (forward declared) is allowed it usually results in nasty compiler errors if the type's destructor is inline (or not defined at all).

Reply:
If `CGUIFontCacheImpl` has a cpp file then the destructor could be defined there, avoiding those nasty errors